theano/gpuarray/tests/check_dnn.sh

```diff
@@ -0,0 +1,74 @@
+set -e
+
+## Get all DNN algorithms supported by Theano.
+
+export SUPPORTED_DNN_CONV_ALGO_FWD=$(python -c "from theano.configdefaults import SUPPORTED_DNN_CONV_ALGO_RUNTIME, SUPPORTED_DNN_CONV_ALGO_FWD; print('\n'.join(SUPPORTED_DNN_CONV_ALGO_FWD + SUPPORTED_DNN_CONV_ALGO_RUNTIME))")
+
+export SUPPORTED_DNN_CONV_ALGO_BWD_DATA=$(python -c "from theano.configdefaults import SUPPORTED_DNN_CONV_ALGO_RUNTIME, SUPPORTED_DNN_CONV_ALGO_BWD_DATA; print('\n'.join(SUPPORTED_DNN_CONV_ALGO_BWD_DATA + SUPPORTED_DNN_CONV_ALGO_RUNTIME))")
+
+export SUPPORTED_DNN_CONV_ALGO_BWD_FILTER=$(python -c "from theano.configdefaults import SUPPORTED_DNN_CONV_ALGO_RUNTIME, SUPPORTED_DNN_CONV_ALGO_BWD_FILTER; print('\n'.join(SUPPORTED_DNN_CONV_ALGO_BWD_FILTER + SUPPORTED_DNN_CONV_ALGO_RUNTIME))")
+
+export SUPPORTED_DNN_CONV_PRECISION=$(python -c "from theano.configdefaults import SUPPORTED_DNN_CONV_PRECISION; print('\n'.join(SUPPORTED_DNN_CONV_PRECISION))")
+
+## List special Theano DNN tests that depend on DNN algorithms.
+## Other Theano DNN tests will be run once.
+## These special Theano DNN tests will be run 
+## for every combination of DNN algorithms.
+
+declare -a special_tests=('theano.gpuarray.tests.test_dnn.test_conv3d_bwd'
+                          'theano.gpuarray.tests.test_dnn.test_conv3d_fwd'
+                          'theano.gpuarray.tests.test_dnn.test_dnn_conv_alpha_output_merge'
+                          'theano.gpuarray.tests.test_dnn.test_dnn_conv_border_mode'
+                          'theano.gpuarray.tests.test_dnn.test_dnn_conv_grad'
+                          'theano.gpuarray.tests.test_dnn.test_dnn_conv_inplace'
+                          'theano.gpuarray.tests.test_dnn.test_dnn_conv_merge'
+                          'theano.gpuarray.tests.test_dnn.TestDnnInferShapes')
+
+## Precompute nosetests arguments used later.
+
+# PB: To test quickly this script, you can temporarly 
+# add "--collect-only" arg into this variable.
+export NOSEARGS="-xvs --collect-only" #export NOSEARGS="-xvs"
+
+# Convert list of special tests to a list of tests to be executed by nosetests.
+export SPECIAL_NOSETESTS=$(echo ${special_tests[@]} | sed 's/\.test_dnn\./.test_dnn:/g')
+
+# Convert list of special tests to a list of tests to be excluded by nosetests.
+export SPECIAL_NOSETESTS_EXCLUDED=$(for test in "${special_tests[@]}"; do echo "--exclude-test=$test"; done)
+
+
+## Run all Theano DNN tests without special tests.
+
+echo Running independent DNN tests.
+
+THEANO_FLAGS=warn.dnn_recent=False eval "nosetests $NOSEARGS $(echo $SPECIAL_NOSETESTS_EXCLUDED) theano/gpuarray/tests/test_dnn.py"
+
+## Run all special Theano DNN tests for every combination of DNN algorithms.
+# This may take a very (very) long time.
+
+echo Running DNN tests depending on algorithms parameters.
+
+for fwd in $SUPPORTED_DNN_CONV_ALGO_FWD
+do
+    for bwd_filter in $SUPPORTED_DNN_CONV_ALGO_BWD_FILTER
+    do
+        for bwd_data in $SUPPORTED_DNN_CONV_ALGO_BWD_DATA
```

Comment:
I don't think we need to test all combinations of algos, they should be pretty independent.